### PR TITLE
Fix grpc is not working with logger

### LINKED ISSUE #2251: logger is not working with grpc request.
/kind bug

I`m following payload logger example with custom preditor using grpc.

when I add logger, grpc request returns error

isvc yaml (when i remove logger, it works well and I`m using triton on custom predictor for testing)

```yaml
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  name: torchscript-cifar10
  annotations:
    prometheus.io/scrape: 'true'
    prometheus.io/port: '8002'
spec:
  predictor:
    serviceAccountName: sa
    logger:
      mode: all
      url: http://message-dumper.default/
    containers:
      - name: triton
        image: nvcr.io/nvidia/tritonserver:20.10-py3
        command:
          - tritonserver
        args:
          - --model-repository=/mnt/models
          - --http-port=8088
          - --grpc-port=9000
        ports:
          - containerPort: 9000
            name: h2c
            protocol: TCP
        env:
          - name: STORAGE_URI
            value: 
        resources:
          limits:
            memory: 10Gi
            nvidia.com/gpu: 1
```

grpc request
```
python client.py
<_InactiveRpcError of RPC that terminated with:
	status = StatusCode.UNKNOWN
	details = "Stream removed"
	debug_error_string = "{"created":"@1655347294.370618675","description":"Error received from peer ipv4:172.16.200.4:30791","file":"src/core/lib/surface/call.cc","file_line":1070,"grpc_message":"Stream removed","grpc_status":2}"
>
Traceback (most recent call last):
  File "client.py", line 48, in <module>
    response = grpc_stub.ServerReady(request)
  File "/usr/local/lib/python3.8/dist-packages/grpc/_channel.py", line 946, in __call__
    return _end_unary_response_blocking(state, call, False, None)
  File "/usr/local/lib/python3.8/dist-packages/grpc/_channel.py", line 849, in _end_unary_response_blocking
    raise _InactiveRpcError(state)
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.UNKNOWN
	details = "Stream removed"
	debug_error_string = "{"created":"@1655347294.373034711","description":"Error received from peer ipv4:172.16.200.4:30791","file":"src/core/lib/surface/call.cc","file_line":1070,"grpc_message":"Stream removed","grpc_status":2}"
```

**Environment:**

- Istio Version: 1.11.5
- Knative Version: 1.3
- Kubeflow version: 0.8.0
- Kubernetes version: (use `kubectl version`): 1.21
- OS (e.g. from `/etc/os-release`): ubuntu 20.04

### PR DESCRIPTION
**Pull request by andyi2it**
Signed-off-by: Andrews Arokiam <andrews.arokiam@ideas2it.com>

**What this PR does / why we need it**:
* Fix grpc inference request fails when used with logger.

Screenshot from 2022-10-27 16-40-21

Was able to replicate that the above error shows up without these changes and was able to verify it works without these errors with the new changes

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2251 

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [x] Ran grpc without the change and with the change and can confirm its working
# With the new changes 
image

# Without the changes
image

There were 2 headers that were missing.

Also, the trailer headers are missing based on which the status is being sent as seen below. This could be the reason why grpc is failing with loggers.
Screenshot from 2022-11-10 10-44-37

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by andyi2it** [comment]
Screenshot from 2022-10-27 16-40-21

Was able to replicate that the above error shows up without these changes and was able to verify it works without these errors with the new changes

**Comment by andyi2it** [comment]
# With the new changes 
image

# Without the changes
image

There were 2 headers that were missing.

Also, the trailer headers are missing based on which the status is being sent as seen below. This could be the reason why grpc is failing with loggers.
Screenshot from 2022-11-10 10-44-37

File: test/scripts/gh-actions/status-check.sh [hunk 1]
```diff
@@ -9,4 +9,6 @@ kubectl get pods -n kserve-ci-e2e-test
 kubectl describe pods -n kserve-ci-e2e-test
 kubectl get events -n kserve-ci-e2e-test
 kubectl logs -l control-plane=kserve-controller-manager -n kserve -c manager
-kubectl logs -l 'component in (predictor, explainer, transfomer)' -c kserve-container -n kserve-ci-e2e-test
\ No newline at end of file
+kubectl logs -l 'component in (predictor, explainer, transfomer)' -c kserve-container -n kserve-ci-e2e-test
```
Comment (yuzisun) [inline test/scripts/gh-actions/status-check.sh]:
```suggestion
kubectl logs -l 'component in (predictor, explainer, transformer)' -c kserve-container -n kserve-ci-e2e-test
```

File: test/scripts/gh-actions/status-check.sh [hunk 2]
```diff
@@ -9,4 +9,6 @@ kubectl get pods -n kserve-ci-e2e-test
...
+echo "list python dependencies"
+pip list
```
Comment (yuzisun) [inline test/scripts/gh-actions/status-check.sh]: Do we still need this?

File: test/e2e/predictor/test_grpc.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                                                      label_selector='serving.kserve.io/inferenceservice={}'.
+                                                      format(msg_dumper))
+    time.sleep(5)
+    log = ''
+    for pod in pods.items:
+        log += kserve_client.core_api.read_namespaced_pod_log(name=pod.metadata.name,
+                                                              namespace=pod.metadata.namespace,
+                                                              container="kserve-container")
+        print(log)
+    assert ("org.kubeflow.serving.inference.request" in log)
```
Comment (yuzisun) [inline test/e2e/predictor/test_grpc.py:125]: hmm seems like we are still using the kubeflow package name

File: test/e2e/predictor/test_grpc.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
+#
```
Comment (yuzisun) [inline test/e2e/predictor/test_grpc.py:2]: missing `# Copyright 2022 The KServe Authors.`

**Pull request merged**